### PR TITLE
Capturing vectorstore deployment url

### PR DESCRIPTION
**Pull request by sachintendulkar576123**
## Proposed changes

This PR:

- we are capturing the vector deployment url for OpenSearch vectorstore

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by oi-raanne — CHANGES_REQUESTED** [review]

File: src/monocle_apptrace/utils.py [hunk 1]
```diff
@@ -218,3 +221,44 @@ def get_workflow_name(span: Span) -> str:
...
+        if hasattr(my_map, 'client') and '_endpoint' in my_map.client.__dict__:
+            return my_map.client.__dict__['_endpoint']
+
+        host, port = get_keys_as_tuple(my_map.__dict__, 'host', 'port')
+        host_port_result = f"{host[0]}:{port[0]}" if host or port else None
+        if host_port_result:
+            return host_port_result
+    return None
+
+def get_keys_as_tuple(dictionary, *keys):
```
Comment (oi-raanne) [inline src/monocle_apptrace/utils.py:249]:
```
def get_keys_as_tuple(dictionary, *keys):
    return tuple(dictionary.get(key, None) for key in keys)
```

File: src/monocle_apptrace/utils.py [hunk 2]
```diff
@@ -218,3 +221,44 @@ def get_workflow_name(span: Span) -> str:
...
+        host = get_host_from_map(my_map, keys_to_check)
+        if host:
+            return host
+
+    else:
+        if hasattr(my_map, 'client') and '_endpoint' in my_map.client.__dict__:
+            return my_map.client.__dict__['_endpoint']
+
+        host, port = get_keys_as_tuple(my_map.__dict__, 'host', 'port')
+        host_port_result = f"{host[0]}:{port[0]}" if host or port else None
```
Comment (oi-raanne) [inline src/monocle_apptrace/utils.py:244]: return host if no port available

File: src/monocle_apptrace/utils.py [hunk 3]
```diff
@@ -218,3 +221,44 @@ def get_workflow_name(span: Span) -> str:
     except Exception as e:
         logger.exception(f"Error getting workflow name: {e}")
         return None
+
+def get_vectorstore_deployment(my_map):
+    if isinstance(my_map, dict):
+        if '_client_settings' in my_map:
+            client = my_map['_client_settings'].__dict__
+            host, port = get_keys_as_tuple(client, 'host', 'port')
+            host_port_result = f"{host[0]}:{port[0]}" if host or port else None
```
Comment (oi-raanne) [inline src/monocle_apptrace/utils.py:230]: return host if no port available

**Comment by oi-raanne** [comment]
unit tests missing

**Review by oi-raanne — CHANGES_REQUESTED** [review]

**Review by oi-raanne — APPROVED** [review]

**Review by kshitiz-okahu — APPROVED** [review]

**Pull request merged**